### ISSUE TITLE
"pip install labml-nn" generated errors. How to resolve it and complete the installation?

### ISSUE THREAD

**Issue by jxwanguab** [labmlai/annotated_deep_learning_paper_implementations#247]
Here is the message:

(base) C:\Windows\system32>pip install labml-nn

WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
Collecting labml-nn
  Downloading labml_nn-0.4.136-py3-none-any.whl (434 kB)
     |████████████████████████████████| 434 kB 6.4 MB/s
Requirement already satisfied: numpy in c:\users\user1\appdata\roaming\python\python38\site-packages (from labml-nn) (1.21.2)
Collecting labml==0.4.168
  Downloading labml-0.4.168-py3-none-any.whl (130 kB)
     |████████████████████████████████| 130 kB ...
Requirement already satisfied: torchvision in d:\users\user1\anaconda3\lib\site-packages (from labml-nn) (0.10.0)
Collecting torchtext
  Downloading torchtext-0.17.1-cp38-cp38-win_amd64.whl (1.9 MB)
     |████████████████████████████████| 1.9 MB ...
Collecting fairscale
  Downloading fairscale-0.4.13.tar.gz (266 kB)
     |████████████████████████████████| 266 kB ...
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
Requirement already satisfied: torch in d:\users\user1\anaconda3\lib\site-packages (from labml-nn) (1.9.0)
Collecting einops
  Downloading einops-0.7.0-py3-none-any.whl (44 kB)
     |████████████████████████████████| 44 kB ...
Collecting labml-helpers==0.4.89
  Downloading labml_helpers-0.4.89-py3-none-any.whl (24 kB)
Requirement already satisfied: pyyaml in d:\users\user1\anaconda3\lib\site-packages (from labml==0.4.168->labml-nn) (6.0)
Collecting gitpython
  Downloading GitPython-3.1.42-py3-none-any.whl (195 kB)
     |████████████████████████████████| 195 kB ...
Collecting numpy
  Downloading numpy-1.24.4-cp38-cp38-win_amd64.whl (14.9 MB)
     |████████████████████████████████| 14.9 MB ...
Requirement already satisfied: typing-extensions in d:\users\user1\anaconda3\lib\site-packages (from torch->labml-nn) (4.7.1)
Collecting gitdb<5,>=4.0.1
  Downloading gitdb-4.0.11-py3-none-any.whl (62 kB)
     |████████████████████████████████| 62 kB ...
Collecting smmap<6,>=3.0.1
  Downloading smmap-5.0.1-py3-none-any.whl (24 kB)
Requirement already satisfied: requests in d:\users\user1\anaconda3\lib\site-packages (from torchtext->labml-nn) (2.31.0)
Requirement already satisfied: tqdm in d:\users\user1\anaconda3\lib\site-packages (from torchtext->labml-nn) (4.65.0)
Collecting torch
  Downloading torch-2.2.1-cp38-cp38-win_amd64.whl (198.6 MB)
     |████████████████████████████████| 198.6 MB 6.4 MB/s
Collecting torchdata==0.7.1
  Downloading torchdata-0.7.1-cp38-cp38-win_amd64.whl (1.3 MB)
     |████████████████████████████████| 1.3 MB ...
Requirement already satisfied: jinja2 in d:\users\user1\anaconda3\lib\site-packages (from torch->labml-nn) (3.1.2)
Requirement already satisfied: filelock in d:\users\user1\anaconda3\lib\site-packages (from torch->labml-nn) (3.9.0)
Requirement already satisfied: fsspec in d:\users\user1\anaconda3\lib\site-packages (from torch->labml-nn) (2023.9.2)
Collecting typing-extensions
  Downloading typing_extensions-4.10.0-py3-none-any.whl (33 kB)
Requirement already satisfied: sympy in d:\users\user1\anaconda3\lib\site-packages (from torch->labml-nn) (1.11.1)
Requirement already satisfied: networkx in d:\users\user1\anaconda3\lib\site-packages (from torch->labml-nn) (3.1)
Requirement already satisfied: urllib3>=1.25 in d:\users\user1\anaconda3\lib\site-packages (from torchdata==0.7.1->torchtext->labml-nn) (1.26.16)
Requirement already satisfied: MarkupSafe>=2.0 in d:\users\user1\anaconda3\lib\site-packages (from jinja2->torch->labml-nn) (2.1.1)
Requirement already satisfied: certifi>=2017.4.17 in d:\users\user1\anaconda3\lib\site-packages (from requests->torchtext->labml-nn) (2024.2.2)
Requirement already satisfied: charset-normalizer<4,>=2 in d:\users\user1\anaconda3\lib\site-packages (from requests->torchtext->labml-nn) (2.0.4)
Requirement already satisfied: idna<4,>=2.5 in d:\users\user1\anaconda3\lib\site-packages (from requests->torchtext->labml-nn) (3.4)
Requirement already satisfied: mpmath>=0.19 in d:\users\user1\anaconda3\lib\site-packages (from sympy->torch->labml-nn) (1.3.0)
Requirement already satisfied: pillow>=5.3.0 in d:\users\user1\anaconda3\lib\site-packages (from torchvision->labml-nn) (8.3.1)
Collecting torchvision
  Downloading torchvision-0.17.1-cp38-cp38-win_amd64.whl (1.2 MB)
     |████████████████████████████████| 1.2 MB ...
Collecting pillow!=8.3.*,>=5.3.0
  Downloading pillow-10.2.0-cp38-cp38-win_amd64.whl (2.6 MB)
     |████████████████████████████████| 2.6 MB 6.4 MB/s
Requirement already satisfied: colorama in d:\users\user1\anaconda3\lib\site-packages (from tqdm->torchtext->labml-nn) (0.4.6)
Building wheels for collected packages: fairscale
  Building wheel for fairscale (PEP 517) ... done
  Created wheel for fairscale: filename=fairscale-0.4.13-py3-none-any.whl size=332118 sha256=56f7a186cb096dbde7f73843232953a22e6ad233f5d0439ffbcb18fc272babcc
  Stored in directory: c:\users\user1\appdata\local\pip\cache\wheels\b8\02\9b\dc7d4ff5145afdd28f456dae6605a46619af0370eca30d8d7e
Successfully built fairscale
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
Installing collected packages: smmap, typing-extensions, gitdb, torch, numpy, gitpython, torchdata, pillow, labml, torchvision, torchtext, labml-helpers, fairscale, einops, labml-nn
  Attempting uninstall: typing-extensions
    WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
    Found existing installation: typing-extensions 4.7.1
    Uninstalling typing-extensions-4.7.1:
      Successfully uninstalled typing-extensions-4.7.1
  Attempting uninstall: torch
    WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
    Found existing installation: torch 1.9.0
    Uninstalling torch-1.9.0:
      Successfully uninstalled torch-1.9.0
  Attempting uninstall: numpy
    Found existing installation: numpy 1.21.2
    Uninstalling numpy-1.21.2:
      Successfully uninstalled numpy-1.21.2
  Attempting uninstall: pillow
    WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
    Found existing installation: Pillow 8.3.1
    Uninstalling Pillow-8.3.1:
      Successfully uninstalled Pillow-8.3.1
  Attempting uninstall: torchvision
    WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
    WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
    Found existing installation: torchvision 0.10.0
    Uninstalling torchvision-0.10.0:
      Successfully uninstalled torchvision-0.10.0
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
visdom 0.1.8.9 requires jsonpatch, which is not installed.
imgaug 0.4.0 requires opencv-python-headless, which is not installed.
albumentations 1.0.3 requires opencv-python-headless>=4.1.1, which is not installed.
tensorflow 2.6.0 requires absl-py~=0.10, but you have absl-py 1.4.0 which is incompatible.
tensorflow 2.6.0 requires numpy~=1.19.2, but you have numpy 1.24.4 which is incompatible.
tensorflow 2.6.0 requires six~=1.15.0, but you have six 1.16.0 which is incompatible.
tensorflow 2.6.0 requires termcolor~=1.1.0, but you have termcolor 2.1.0 which is incompatible.
tensorflow 2.6.0 requires typing-extensions~=3.7.4, but you have typing-extensions 4.10.0 which is incompatible.
tensorflow 2.6.0 requires wrapt~=1.12.1, but you have wrapt 1.14.1 which is incompatible.
tensorflow-gpu 2.6.0 requires absl-py~=0.10, but you have absl-py 1.4.0 which is incompatible.
tensorflow-gpu 2.6.0 requires numpy~=1.19.2, but you have numpy 1.24.4 which is incompatible.
tensorflow-gpu 2.6.0 requires six~=1.15.0, but you have six 1.16.0 which is incompatible.
tensorflow-gpu 2.6.0 requires termcolor~=1.1.0, but you have termcolor 2.1.0 which is incompatible.
tensorflow-gpu 2.6.0 requires typing-extensions~=3.7.4, but you have typing-extensions 4.10.0 which is incompatible.
tensorflow-gpu 2.6.0 requires wrapt~=1.12.1, but you have wrapt 1.14.1 which is incompatible.
tensorboard 2.10.0 requires protobuf<3.20,>=3.9.2, but you have protobuf 3.20.3 which is incompatible.
pytorch-lightning 1.3.8 requires PyYAML<=5.4.1,>=5.1, but you have pyyaml 6.0 which is incompatible.
Successfully installed einops-0.7.0 fairscale-0.4.13 gitdb-4.0.11 gitpython-3.1.42 labml-0.4.168 labml-helpers-0.4.89 labml-nn-0.4.136 numpy-1.24.4 pillow-10.2.0 smmap-5.0.1 torch-2.2.1 torchdata-0.7.1 torchtext-0.17.1 torchvision-0.17.1 typing-extensions-4.10.0
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -5py (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -umpy (d:\users\user1\anaconda3\lib\site-packages)
WARNING: Ignoring invalid distribution -ip (d:\users\user1\anaconda3\lib\site-packages)

(base) C:\Windows\system32>
